ambry-coordinator/src/test/java/com.github.ambry.coordinator/MockBlockingChannel.java

```diff
@@ -106,9 +107,12 @@ public void send(Send send)
         BlobId blobId = putRequest.getBlobId();
         BlobProperties blobProperties = putRequest.getBlobProperties();
         ByteBuffer userMetadata = putRequest.getUsermetadata();
-        BlobOutput blobOutput = new BlobOutput(putRequest.getBlobSize(), putRequest.getBlobStream());
+        // Since this is a test class, we assume that the Inputstream sent in via PutRequest is ByteBufferInputStream
```

Comment:
Why care about the extra copy in the tests and add this comment? Just create a `ByteBufferInputStream` and keep things simple?